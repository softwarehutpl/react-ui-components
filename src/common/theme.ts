const theme: { [key: string]: { [key: string]: { [key: string]: string } } } = {
  colors: {
    primary: {
      base: '#000000',
      light: '#ffffff',
      dark: '#a8b3bd',
    },
    secondary: {
      base: '#3690e3',
      light: '#75b0e6',
      dark: '#034d91',
    },
    error: {
      base: '#f44336',
      light: '#f56c62',
      dark: '#a1140a',
    },
    warning: {
      base: '#faa72d',
      light: '#ffc36b',
      dark: '#995b00',
    },
    info: {
      base: '#7979d4',
      light: '#9797d1',
      dark: '#4c4cc7',
    },
    success: {
      base: '#4caf50',
      light: '#88c28a',
      dark: '#124f14',
    },
    disabled: {
      base: '#d9d9d9',
      light: '#f0eded',
      dark: '#8c8989',
    },
  },
  // TODO: change it in dedicated task for that (change ease-out for example and add more options)
  transitions: {
    all: {
      fast: 'all 0.5s ease-out',
      mid: 'all 2s ease-out',
      slow: 'all 4s ease-out',
    },
    background: {
      fast: 'background-color 0.5s ease-out',
      mid: 'background-color 2s ease-out',
      slow: 'background-color 4s ease-out',
    },
    fontColor: {
      fast: 'color 0.5s ease-out',
      mid: 'color 2s ease-out',
      slow: 'color 4s ease-out',
    }
  }
};

export default theme;
