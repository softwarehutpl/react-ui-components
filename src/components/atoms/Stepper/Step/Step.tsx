import React from 'react';
import styled from 'styled-components';
import stepImage from '../../../../assets/register.svg';

export interface WrapperProps {
  width?: string;
  height?: string;
  color?: any;
  isActive?: boolean;
};

export interface IconProps {
  isActive?: boolean;
  color?: any;
};

export interface TitleProps {
  isActive?: boolean;
};

export interface StepProps extends WrapperProps, IconProps, TitleProps {
  stepClassName?: string;
  imageClassName?: string;
  image? : any;
  title?: string;
};

export const Wrapper = styled.div<WrapperProps>`
  position: relative; 
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: ${({ width }) => width};
  height: ${({ height }) => height};

  &:after {
    content: " ";
    position: absolute; 
    left: 50%;
    top: calc(50% + 35px);
    width: 100%;
    height: 3px;
    background-color: ${({ color, theme }) => (theme.colors[color].base)};
    z-index: -1;
  }
  &:last-child:after {
    display: none;
  }
`;

export const Image = styled.img`
  width: 112px;
  height: 90px;
  fill: red;
`;

export const Icon = styled.div<IconProps>`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  border-color: ${( { theme, color }) => (theme.colors[color].base)};
  border-width: 3px;
  border-style: solid;
  background-color: ${({ color, theme, isActive }) => (isActive ? theme.colors[color].base : '#fff')};
    &:after {
      content: '';
      position: absolute;
      top: 9px;
      color: #fff;
      width: 20px;
      height: 30px;
      border: solid  #fff;
      border-width: 0 2px 2px 0;
      transform: rotate(45deg);
    }
`;

export const Title = styled.span<TitleProps>`
  font-weight: ${({ isActive }) =>  (isActive  && 500)};
`;

const Step: React.StatelessComponent<StepProps> = ({ 
  stepClassName,
  imageClassName,
  image, 
  title,
  isActive,
  color,
  width,
  height,
}) => (
  <Wrapper
    className={stepClassName}
    color={color}
    width={width}
    height={height}
    isActive={isActive}
  >
    <Image
      src={image ? image : stepImage}
      alt="step-image"
      className={imageClassName}
    />
    <Icon
      color={color}
      isActive={isActive}
    />
    <Title 
      isActive={isActive}
    >
      {title}
    </Title>
  </Wrapper>
);

Step.defaultProps = {
  stepClassName: '',
  color: 'primary',
  width: '100%',
  height: '300px',
  isActive: true,
};

export default Step;
