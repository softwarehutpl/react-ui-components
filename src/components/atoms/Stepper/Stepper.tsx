import React from 'react';

export const Stepper: React.StatelessComponent<{}> = () => (
  <div className="">Stepper
  </div>
);
