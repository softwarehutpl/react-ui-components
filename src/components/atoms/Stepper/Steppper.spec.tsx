import React from 'react';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import Stepper, { StepperContainer }from './Stepper';;
import Step from './Step/Step';
import { ThemeProvider } from 'styled-components';
import theme from '../../../common/theme';

describe('<Stepper', () => {
  let component: any;
  it('should render corectly', () => {
    component = create(
      <ThemeProvider theme={theme}>
        <Stepper  items={[{title: 'test'}]}  color="primary"  activeIndex={0}/>
      </ThemeProvider>
      );
    expect(component).toMatchSnapshot();
  });
});
