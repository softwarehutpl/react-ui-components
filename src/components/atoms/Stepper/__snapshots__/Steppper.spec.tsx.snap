// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stepper should render corectly 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 300px;
}

.c1:after {
  content: " ";
  position: absolute;
  left: 50%;
  top: calc(50% + 35px);
  width: 100%;
  height: 3px;
  background-color: #000000;
  z-index: -1;
}

.c1:last-child:after {
  display: none;
}

.c2 {
  width: 112px;
  height: 90px;
  fill: red;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  border-color: #000000;
  border-width: 3px;
  border-style: solid;
  background-color: #000000;
}

.c3:after {
  content: '';
  position: absolute;
  top: 9px;
  color: #fff;
  width: 20px;
  height: 30px;
  border: solid #fff;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c4 {
  font-weight: 500;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="primary"
    height="300px"
    width="100%"
  >
    <img
      alt="step-image"
      className="c2"
      src="register.svg"
    />
    <div
      className="c3"
      color="primary"
    />
    <span
      className="c4"
    >
      test
    </span>
  </div>
</div>
`;
